docs/standard/parallel-programming/snippets/cs/DownloadCache.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+public static class DownloadCache
+{
+    private static readonly ConcurrentDictionary<string, string> s_cachedDownloads = new();
+    private static readonly HttpClient s_httpClient = new();
+
+    public static Task<string> DownloadStringAsync(string address)
+    {
+        if (s_cachedDownloads.TryGetValue(address, out string content))
+        {
+            return Task.FromResult<string>(content);
```

Comment:
I expect this to cause the "Name can be simplified" analyzer to issue a warning.

```suggestion
            return Task.FromResult(content);
```